indexing-service/src/test/java/io/druid/indexing/common/TestMergeTask.java

```diff
@@ -1,90 +0,0 @@
-/*
- * Licensed to Metamarkets Group Inc. (Metamarkets) under one
- * or more contributor license agreements. See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership. Metamarkets licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License. You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied. See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-
-package io.druid.indexing.common;
-
-import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonProperty;
-import com.fasterxml.jackson.annotation.JsonTypeName;
-import com.google.common.collect.Lists;
-import io.druid.indexing.common.task.MergeTask;
-import io.druid.query.aggregation.AggregatorFactory;
-import io.druid.segment.IndexSpec;
-import io.druid.timeline.DataSegment;
-import org.joda.time.DateTime;
-import org.joda.time.Interval;
-
-import java.util.List;
-
-/**
- */
-@JsonTypeName("test")
-public class TestMergeTask extends MergeTask
```

Comment:
It made it difficult to add runtime location checking stuff to the WorkerTaskMonitor unit tests, because of how weird this task is (notice it returns `TaskStatus.running`… that's arguably broken, `Task.run` should only be returning "complete" statuses like success/failure). So I replaced it with two new ones, "immediateSuccess" and "unending" that behave better.

And while doing that I also swapped out the other usages because it bugged me that this task was so weird.